Automattic/jetpack
# Protect: Use Blog ID in links to WPCOM instead of site slug.

## Proposed changes:
Adjust checkout links in Protect plugin to use blog ID instead of slug for multi-URL support.

### Other information:


## Jetpack product discussion
pf5801-93-p2

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
To test the blog, you'll need two domains pointing to the same site. One way to do that is to setup two proxying tunnels to the same local environment.

**Connect Jetpack with a free plan on URL A, and load it via URL B.**

If your site falls into IDC due to concurrency issues with IDC secret validation (fix pending), remove `jetpack_sync_error_idc` option.

1. Using URL B (not the main cache site URL), go to "Jetpack -> Protect" and click the "Get paid plan" button. Confirm you're taken to connection flow and then checkout. Don't finish the checkout yet.
2. Go back to "Jetpack -> Protect" and get a free plan.
3. Check the following "Upgrade" links to confirm they all lead to checkout with URL A (main cache site URL) in the checkout page URL:
    - "Scan" tab -> Choose a threat -> "How to fix"
    - "Scan" tab -> "Advanced scan results" in the footer
    - "Firewall" tab -> "Upgrade to update automatic security rules"
    - "Firewall" tab -> "Enable automatic firewall protection" -> "Upgrade to keep your site secure..."
4. Purchase the paid plan using URL B. You should be redirected back to "Jetpack -> Protect", but on the URL A.
5. Go to "Scan" tab -> "Go to Cloud" and check the link to Jetpack Cloud via URL B. Confirm it leads to proper Jetpack Cloud page with cache site detected correctly (correct site name in the left sidebar).
